#163556785: implement endpoint for updating party name

Bobjayafam opened the pull request:
#### What does this PR do?
create an endpoint for admin to update the name of a specific party
#### Description of Task to be completed?
-  add update one part method to party controller
- write tests for this endpoint
#### How should this be manually tested?
- clone the GitHub repo
- checkout branch ``ft-admin-edit-party-name-163556785``
- run ``npm install`` and ``npm run``
- on POSTMAN, send a PATCH request to ``/api/v1/parties/:id/name``., Send the request body to contain the new party name. The response should be a successful operation message.
#### Any background context you want to provide?
N/A
#### What are the relevant pivotal tracker stories?
https://www.pivotaltracker.com/story/show/163556785

Bobjayafam committed:
feat(server): implement endpoint for updating party name

-  add update one part method to party controller
- write tests for this endpoint

[Finishes #163556785]

Bobjayafam commented:
@akinwale-habibullah, kindly review

akinwale-habibullah commented:
Hi @Bobjayafam 

You have two parties with the same id;

![image](https://user-images.githubusercontent.com/24890672/52045762-78baf400-2545-11e9-8c51-ceaf2fa2a92f.png)

When I try to edit party with id = 1, here's what I get;

![image](https://user-images.githubusercontent.com/24890672/52045855-a7d16580-2545-11e9-9ae4-b85df9168779.png)


I advise you change the party named All Progressives Congress, specifically the *ID*

Bobjayafam committed:
feat(server): implement feedback

- change the id of one of two conflicting parties in parties model file

[Finishes #163556785]

Bobjayafam commented:
@akinwale-habibullah I have corrected it. thanks
